### DISCUSSION TITLE
Include changes made by a project script in the dependency bump PR

### DISCUSSION THREAD

**Discussion by gitdallas** [renovatebot/renovate#17688] [Request Help]
When we update our dependencies manually, after the yarn.lock file is updated we run a script that creates a project_manifest file and include that with the dependency bump PR.

An existing github-action will check that if yarn.lock is updated, that project_manifest file is also updated and fail the CI if that isn't the case.

I'm trying to figure out a way to include the project_manifest changes in renovate's bump PRs. I see there are some `postUpgrade` options, but these seem to require self-hosting for which I'm not seeing good instruction on doing (to be fair - I'm new to working with github bots and actions) and it also seems like this might only apply to windows machines based on that the documentation for [Installing and Onboarding](https://docs.renovatebot.com/getting-started/installing-onboarding/#repository-installation) only mentions "Self-hosting on windows".

I've attempted to have a new github action that runs on a new PR by renovatebot and updates project_manifest and pushes the changes to the PR... but I'm having issues with permission being denied to github-actions[bot]. In github my `project/settings/actions - workflow permissions` are "read and write" with "allow github actions to create and approve pull requests", using the secrets.GITHUB_TOKEN in the workflow yml.

I may be wrong, but this seems like something that would be fairly common.. so I'm surprised I can't find an example somewhere.

**Comment by viceice** [comment]
you can self host on any system, we prefer Linux with docker, you can also run on windows and MacOS.

we've also a GitHub action for self hosting


https://github.com/renovatebot/github-action

for configuring GitHub actions you should read the official docs
